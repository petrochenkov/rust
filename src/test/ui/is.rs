// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

#![warn(unused)]

fn main() {
    let my_opt = Some(15);
    if my_opt is Some(x) &&
       x > 10 { //~ ERROR use of possibly uninitialized variable: `x`
        println!("{:?}", x); //~ ERROR use of possibly uninitialized variable: `x`
    }

    let mut iter = my_opt.into_iter();
    while iter.next() is Some(mut y) &&
          y > 10 { //~ ERROR use of possibly uninitialized variable: `y`
        println!("{:?}", y); //~ ERROR use of possibly uninitialized variable: `y`
    }

    let z = '6';
    let is_digit = z is '0' ... '9' && z > '5';
    if is_digit {
        println!("{:?}", z);
    }

    let _ = my_opt is Some(a); //~ WARN variable `a` is assigned to, but never used
                               //~^ WARN value assigned to `a` is never read

    // Variables bound in `is` live until the end of the block in this configuration
    let _ = x;
    let _ = y;
    let _ = z;
}
