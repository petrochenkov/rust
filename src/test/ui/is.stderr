warning: variable `a` is assigned to, but never used
  --> $DIR/is.rs:32:28
   |
32 |     let _ = my_opt is Some(a); //~ WARN variable `a` is assigned to, but never used
   |                            ^
   |
note: lint level defined here
  --> $DIR/is.rs:11:9
   |
11 | #![warn(unused)]
   |         ^^^^^^
   = note: #[warn(unused_variables)] implied by #[warn(unused)]
   = note: to avoid this warning, consider using `_a` instead

warning: value assigned to `a` is never read
  --> $DIR/is.rs:32:13
   |
32 |     let _ = my_opt is Some(a); //~ WARN variable `a` is assigned to, but never used
   |             ^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/is.rs:11:9
   |
11 | #![warn(unused)]
   |         ^^^^^^
   = note: #[warn(unused_assignments)] implied by #[warn(unused)]

error[E0381]: use of possibly uninitialized variable: `x`
  --> $DIR/is.rs:16:8
   |
16 |        x > 10 { //~ ERROR use of possibly uninitialized variable: `x`
   |        ^ use of possibly uninitialized `x`

error[E0381]: use of possibly uninitialized variable: `x`
  --> $DIR/is.rs:17:26
   |
17 |         println!("{:?}", x); //~ ERROR use of possibly uninitialized variable: `x`
   |                          ^ use of possibly uninitialized `x`

error[E0381]: use of possibly uninitialized variable: `y`
  --> $DIR/is.rs:22:11
   |
22 |           y > 10 { //~ ERROR use of possibly uninitialized variable: `y`
   |           ^ use of possibly uninitialized `y`

error[E0381]: use of possibly uninitialized variable: `y`
  --> $DIR/is.rs:23:26
   |
23 |         println!("{:?}", y); //~ ERROR use of possibly uninitialized variable: `y`
   |                          ^ use of possibly uninitialized `y`

error: aborting due to 4 previous errors

